### PR TITLE
Introduce usePresetFilters and useRowSelction hooks

### PR DESCRIPTION
**Pull request by poulch**
closes its introduce new hooks to manage datagrid row selection and filter presets 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](/.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

**Review by andrzejewsky — APPROVED** [review]

**Review by andrzejewsky — DISMISSED** [review]

File: src/hooks/useFilterPresets/useFilterPresets.test.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    // Act
+    result.current.onPresetChange(1);
+
+    // Assert
+    expect(mockNavigate).toHaveBeenCalledWith(
+      `${baseUrl}?${savedPreset.data}&activeTab=1`,
+    );
+  });
+
+  it("should handle prset delete and navigate to base url when active preset is equal deleting preset", () => {
```
Comment (krzysztofzuraw) [inline src/hooks/useFilterPresets/useFilterPresets.test.ts:105]:
```suggestion
  it("should handle preset delete and navigate to base url when active preset is equal deleting preset", () => {
```

File: src/hooks/useFilterPresets/useFilterPresets.test.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    result.current.setPresetIdToDelete(1);
+    result.current.onPresetDelete();
+
+    // Assert
+    expect(mockDeleteStorage).toHaveBeenCalledWith(1);
+    expect(mockReset).toHaveBeenCalled();
+    expect(mockNavigate).toHaveBeenCalledWith(baseUrl);
+  });
+
+  it("should handle prset delete and navigate to active preset when preset to delete is different that preset to delete", () => {
```
Comment (krzysztofzuraw) [inline src/hooks/useFilterPresets/useFilterPresets.test.ts:137]:
```suggestion
  it("should handle preset delete and navigate to active preset when preset to delete is different that preset to delete", () => {
```

File: src/hooks/useFilterPresets/useFilterPresets.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import { stringify } from "qs";
+import { useState } from "react";
+
+export const useFilterPresets = ({
+  params,
+  reset,
+  storageUtils,
+  getUrl,
+}: {
+  params: Record<string, unknown>;
```
Comment (krzysztofzuraw) [inline src/hooks/useFilterPresets/useFilterPresets.ts:18]: **question**: can we type `params` more strictly?

File: src/hooks/useFilterPresets/useFilterPresets.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      navigate(
+        baseUrl.endsWith("?")
+          ? baseUrl + stringify(currentParams)
+          : baseUrl + "?" + stringify(currentParams),
+      );
+    }
+  };
+
+  const onPresetSave = (data: SaveFilterTabDialogFormData) => {
+    const { paresedQs } = prepareQs(location.search);
```
Comment (krzysztofzuraw) [inline src/hooks/useFilterPresets/useFilterPresets.ts:76]:
```suggestion
    const { parsedQs } = prepareQs(location.search);
```

**Review by krzysztofzuraw — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**